### PR TITLE
Use Task in ConstraintBuilder2D

### PR DESCRIPTION
**Pull request by gaschler**

### CODE REVIEW

File: cartographer/mapping/internal/constraints/constraint_builder_2d.cc [hunk 1]
```diff
@@ -77,106 +80,112 @@ void ConstraintBuilder2D::MaybeAddConstraint(
...
+  kQueueLengthMetric->Set(constraints_.size());
+  auto* const constraint = &constraints_.back();
+  const auto* scan_matcher =
+      DispatchScanMatcherConstruction(submap_id, submap->grid());
+  auto constraint_task = common::make_unique<common::Task>();
+  constraint_task->SetWorkItem([=]() EXCLUDES(mutex_) {
+    ComputeConstraint(submap_id, submap, node_id, false, /* match_full_submap */
+                      constant_data, initial_relative_pose, *scan_matcher,
+                      constraint);
+  });
```
Comment (pifon2a) [inline cartographer/mapping/internal/constraints/constraint_builder_2d.cc]: that's a weird formatting, is it with clang-format?

Comment (gaschler) [reply cartographer/mapping/internal/constraints/constraint_builder_2d.cc]: Yeah, it's clang-formatted.

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 2.

**Pull request merged**